stan/math/prim/meta/require_helpers.hpp

```diff
@@ -0,0 +1,401 @@
+ */
+template <class... Checks>
+using require_any_not_t
+    = std::enable_if_t<!math::conjunction<Checks...>::value>;
+
+/** \ingroup macro_helpers
+ * Adds Unary require aliases.
+ * @param check_type The name of the type to check, used to define
+ * `require_<check_type>_t`.
+ * @param checker A struct that returns holds a boolean `value`
+ * @param doxygen_group The doxygen group to add this requires to.
+ */
+#define STAN_ADD_REQUIRE_UNARY(check_type, checker, doxygen_group)       \
+  /*! \ingroup doxygen_group */                                          \
+  /*! \defgroup check_type##_types check_type  */                        \
+  /*! \addtogroup check_type##_types */                                  \
+  /*! @{ */                                                              \
+  /*! \brief Require type satisfies checker */                           \
+  template <typename T>                                                  \
+  using require_##check_type##_t = require_t<checker<std::decay_t<T>>>;  \
+  /*! \brief Require type does not satisfy checker */                    \
+  template <typename T>                                                  \
+  using require_not_##check_type##_t                                     \
+      = require_not_t<checker<std::decay_t<T>>>;                         \
+  /*! \brief Require all of the types satisfy checker */                 \
+  template <typename... Types>                                           \
+  using require_all_##check_type##_t                                     \
+      = require_all_t<checker<std::decay_t<Types>>...>;                  \
+  /*! \brief Require any of the types satisfy checker */                 \
+  template <typename... Types>                                           \
+  using require_any_##check_type##_t                                     \
+      = require_any_t<checker<std::decay_t<Types>>...>;                  \
+  /*! \brief Require none of the types satisfy checker */                \
+  template <typename... Types>                                           \
+  using require_all_not_##check_type##_t                                 \
+      = require_all_not_t<checker<std::decay_t<Types>>...>;              \
+  /*! \brief Require at least one of the types do not satisfy checker */ \
+  template <typename... Types>                                           \
+  using require_any_not_##check_type##_t                                 \
+      = require_any_not_t<checker<std::decay_t<Types>>...>;              \
+/*! @} */
+
+/** \ingroup macro_helpers
+ * Adds unary require aliases that check the `scalar_type`.
+ * @param check_type The name of the type to check, used to define
+ * `require_<check_type>_t`.
+ * @param checker A struct that returns holds a boolean `value`
+ * @param doxygen_group The doxygen group to add this requires to.
+ */
+#define STAN_ADD_REQUIRE_UNARY_SCALAR(check_type, checker, doxygen_group)  \
```

Comment:
If I understand the code correctly this is used to generate requires we did not have before this PR. This is fine, I just dont like how they are named. We now have `require_eigen_vt` and `require_var_vt`. In one case the check works on container and in other on its value. I think we should change the names of the new ones. Maybe `require_vt_var`?

Also `STAN_ADD_REQUIRE_UNARY_SCALAR` and `STAN_ADD_REQUIRE_UNARY_VALUE` are always used together. You can replace them with a single macro that does both.

Reply:
> require_vt_var

I like it! In the past I've seen a few places where we accept any scalar/container type as long as it contains a var etc.  

> Also STAN_ADD_REQUIRE_UNARY_SCALAR and STAN_ADD_REQUIRE_UNARY_VALUE are always used together. You can replace them with a single macro that does both.

Will do